libdevice/device_imf.hpp

```diff
@@ -488,6 +488,77 @@ static inline double __trunc(double x) {
 #endif
 }
 
+static inline float __fast_exp10f(float x) {
+#if defined(__LIBDEVICE_HOST_IMPL__)
+  return __builtin_expf(__builtin_logf(10.f) * x);
+#elif defined(__SPIR__)
+  return __spirv_ocl_native_exp10(x);
+#endif
+}
+
+static inline float __fast_expf(float x) {
+#if defined(__LIBDEVICE_HOST_IMPL__)
+  return __builtin_expf(x);
+#elif defined(__SPIR__)
+  return __spirv_ocl_native_exp(x);
+#endif
+}
+
+static inline float __fast_logf(float x) {
+#if defined(__LIBDEVICE_HOST_IMPL__)
+  return __builtin_logf(x);
+#elif defined(__SPIR__)
+  return __spirv_ocl_native_log(x);
+#endif
+}
+
+static inline float __fast_log2f(float x) {
+#if defined(__LIBDEVICE_HOST_IMPL__)
+  return __builtin_log2f(x);
+#elif defined(__SPIR__)
+  return __spirv_ocl_native_log2(x);
+#endif
+}
+
+static inline float __fast_log10f(float x) {
+#if defined(__LIBDEVICE_HOST_IMPL__)
+  return __builtin_log10f(x);
+#elif defined(__SPIR__)
+  return __spirv_ocl_native_log10(x);
+#endif
+}
+
+static inline float __fast_powf(float x, float y) {
+#if defined(__LIBDEVICE_HOST_IMPL__)
+  return __builtin_powf(x, y);
+#elif defined(__SPIR__)
+  return __spirv_ocl_native_powr(x, y);
+#endif
+}
+
+static inline float __fast_fdividef(float x, float y) {
+  unsigned ybits = __builtin_bit_cast(unsigned, y);
```

Comment:
Hi, @akolesov-intel and @zettai-reido 
For __nv_fast_fdividef: https://docs.nvidia.com/cuda/libdevice-users-guide/__nv_fast_fdividef.html#__nv_fast_fdividef
NV has requirements for 2^126 < y < 2^128 which sycl native math doesn't have, the code below is to handle this.

Reply:
Looks good to me. We might optimize it in future updates: simplify the range check and use fast approximation while fdividef allows 2 ulp rather than correctly rounded x/y.